src/Grpc.AspNetCore.Server/GrpcServiceOptions.cs

```diff
@@ -38,6 +44,28 @@ public class GrpcServiceOptions
         /// Detailed error messages include details from exceptions thrown on the server.
         /// </summary>
         public bool? EnableDetailedErrors { get; set; } = null;
+
+        /// <summary>
+        /// Gets or sets the default compression algorithm used to compress messages sent from the server.
+        /// The grpc-encoding header value in a request will override this setting.
+        /// </summary>
+        public string DefaultCompressionAlgorithm { get; set; }
```

Comment:
Exposing DefaultCompressionAlgorithm makes sense as it's also exposed by C++ server, 
but I think the recommendation is to use DefaultCompressionLevel instead, because it allows the server to choose the compression algorithm intelligently based on what the client supports (and announces in  grpc-accept-encoding). 
https://github.com/grpc/grpc/blob/bf7ed7a053dc799419f8ff7b278f3af347c309fe/include/grpcpp/server_builder.h#L184

I think its fine to add that functionality later though, exposing DefaultCompressionAlgorithm seems good enough for now and it is not incorrect.

Besides that, the C++ server also allows configuring the compression level at the per-rpc level (in ServerContext) and per-message. I think for now we don't need to support those.
The important parts is to be able to read compressed request, comply with the spec when responding and to be able to disable compression on per-message basis (using WriteFlags.NoCompress, as that's important for security reasons) - and this PR mostly has those.

Reply:
The compression level can be specified when you setup the compression provider for the algorithm. For example, you would configure a gzip compression provider and set the compression level for it then.

Reply:
On this name, DefaultCompressionAlgorithm no longer seems the right name. Now the server will only compress messages when the request grpc-accept-endoing and DefaultCompressionAlgorithm match. Should this be renamed to ResponseCompressionAlgorithm?

Reply:
I have:

* Renamed DefaultCompressionAlgorithm to ResponseCompressionAlgorithm
* Added ResponseCompressionLevel